framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -137,11 +144,15 @@ MeshDiagnosticsGenerator::generate()
   // This deliberately does not trust the mesh to know whether it's already prepared or not
   mesh->prepare_for_use();
 
+  // convert BoundaryNames to IDs and sort
+  _watertight_boundaries = prepareBoundaries(mesh);
+  //_watertight_boundaries = mesh->getBoundaryIDs(_watertight_boundary_names);
```

Comment:
```suggestion
  _watertight_boundaries = mesh->getBoundaryIDs(_watertight_boundary_names);
```
the first one has a general name but only applies to this one task in this one diagnostics